src/sdk/pynni/nni/pbt_tuner/pbt_tuner.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+pbt_tuner.py
+"""
+
+import copy
+import logging
+import os
+import numpy as np
+
+import nni
+from nni.tuner import Tuner
+from nni.utils import NodeType, OptimizeMode, extract_scalar_reward, split_index
+
+import nni.parameter_expressions as parameter_expressions
+
+logger = logging.getLogger('pbt_tuner_AutoML')
+
+experiment_id = os.getenv('NNI_EXP_ID')
+all_ckpt_dir = os.path.join("~/nni/checkpoint/", experiment_id)
```

Comment:
suggest to move these two lines to `__init__` of PBTTuner. Set its input argument `all_checkpoint_dir` to be None, and assign this default value if `all_checkpoint_dir` is None.

Reply:
solved